graphql-jpa-query-schema/src/main/java/com/introproventures/graphql/jpa/query/schema/impl/IntrospectionUtils.java

```diff
@@ -24,25 +33,38 @@ public static CachedIntrospectionResult introspect(Class<?> entity) {
     }
 
     public static boolean isTransient(Class<?> entity, String propertyName) {
-        return isAnnotationPresent(entity, propertyName, Transient.class);
+        if(!introspect(entity).hasPropertyDescriptor(propertyName)) {
+            throw new RuntimeException(new NoSuchFieldException(propertyName));
+        }
+        
+        return Stream.of(isAnnotationPresent(entity, propertyName, Transient.class),
+                         isModifierPresent(entity, propertyName, Modifier::isTransient))
+                     .anyMatch(it -> it.isPresent() && it.get() == true);
     }
-
+    
     public static boolean isIgnored(Class<?> entity, String propertyName) {
-        return isAnnotationPresent(entity, propertyName, GraphQLIgnore.class);
+        return isAnnotationPresent(entity, propertyName, GraphQLIgnore.class)
+                 .orElseThrow(() -> new RuntimeException(new NoSuchFieldException(propertyName)));

```

Comment:
The `orElse(false)` was added before to make sure that the application can start even if the field is not found. There may be a case that we have a getter that is not directly connected with any of the class' properties but is still mapped to the column in the database e.g.
```java
public boolean isValuePresent(){
    return value != null;
}
```
But you are right that we should probably throw this error anyway and try to deal with such failure on the higher level. What do you think?